pkg/controller/status/controller.go

```diff
@@ -0,0 +1,390 @@
+		s.reconcileOnClusterChange()
+	})
+
+	s.worker.Run(stopChan)
+
+	// Ensure all goroutines are cleaned up when the stop channel closes
+	go func() {
+		<-stopChan
+		s.informer.Stop()
+		s.clusterDeliverer.Stop()
+	}()
+}
+
+// Check whether all data stores are in sync. False is returned if any of the informer/stores is not yet
+// synced with the corresponding api server.
+func (s *FederationStatusController) isSynced() bool {
+	if !s.informer.ClustersSynced() {
+		glog.V(2).Infof("Cluster list not synced")
+		return false
+	}
+	if !s.templateController.HasSynced() {
+		glog.V(2).Infof("Template not synced")
+		return false
+	}
+	if !s.statusController.HasSynced() {
+		glog.V(2).Infof("Status not synced")
+		return false
+	}
+
+	clusters, err := s.informer.GetReadyClusters()
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("Failed to get ready clusters: %v", err))
+		return false
+	}
+	if !s.informer.GetTargetStore().ClustersSynced(clusters) {
+		return false
+	}
+	return true
+}
+
+// The function triggers reconciliation of all target federated resources.
+func (s *FederationStatusController) reconcileOnClusterChange() {
+	if !s.isSynced() {
+		s.clusterDeliverer.DeliverAt(allClustersKey, nil, time.Now().Add(s.clusterAvailableDelay))
+	}
+	for _, obj := range s.templateStore.List() {
+		qualifiedName := util.NewQualifiedName(obj.(pkgruntime.Object))
+		s.worker.EnqueueWithDelay(qualifiedName, s.smallDelay)
+	}
+}
+
+func (s *FederationStatusController) reconcile(qualifiedName util.QualifiedName) util.ReconciliationStatus {
+	if !s.isSynced() {
+		return util.StatusNotSynced
+	}
+
+	templateKind := s.typeConfig.GetTemplate().Kind
+	statusKind := s.typeConfig.GetStatus().Kind
+	key := qualifiedName.String()
+
+	glog.V(4).Infof("Starting to reconcile %v %v", statusKind, key)
+	startTime := time.Now()
+	defer glog.V(4).Infof("Finished reconciling %v %v (duration: %v)", statusKind, key, time.Now().Sub(startTime))
+
+	template, err := s.objFromCache(s.templateStore, templateKind, key)
+	if err != nil {
+		return util.StatusError
+	}
+
+	if template == nil || template.GetDeletionTimestamp() != nil {
+		glog.V(4).Infof("No template for %v %v found", templateKind, key)
+		// Status object is removed by GC. So we don't have to do anything more here.
+		return util.StatusAllOK
+	}
+
+	clusterNames, err := s.clusterNames()
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("Failed to get cluster list: %v", err))
+		return util.StatusNotSynced
+	}
+
+	clusterStatus, err := s.clusterStatuses(clusterNames, key)
+	if err != nil {
+		return util.StatusError
+	}
+
+	existingStatus, err := s.objFromCache(s.statusStore, statusKind, key)
+	if err != nil {
+		return util.StatusError
+	}
+
+	federatedResource := util.FederatedResource{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       s.typeConfig.GetStatus().Kind,
+			APIVersion: fmt.Sprintf("%s/%s", s.typeConfig.GetStatus().Group, s.typeConfig.GetStatus().Version),
```

Comment:
For core v1 resources, which have an implicit group, this formatting scheme will result in `/v1`.  I suggest using the `schema.GroupVersion` type to ensure safety:

https://github.com/kubernetes-sigs/federation-v2/blob/master/vendor/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L179

Reply:
Updated now. Thanks for catching this issue.